src/csstree/BlockSplitter.py

```diff
@@ -0,0 +1,233 @@
+from __future__ import print_function
+import re
+import logging
+import logging_config #to configure logging, no calls needed
+from CssTree.CssElement import CssElement
+
+
+
+MIN_ZOOM = 1
+MAX_ZOOM = 19
+
+
+BLOCK_SPLITTER = re.compile(r'([^@{]*)\s*\{(.*?)\}', re.DOTALL | re.MULTILINE)
+ZOOM = re.compile(r'(.*?)(\|z[\d\-]*?)?(\[.*)?') #deprecated
+ONE_ZOOM = re.compile(r'(\d{1,2})$')
+ZOOM_RANGE = re.compile(r'(\d{1,2})-(\d{1,2})$')
+ZOOM_TO_MAX = re.compile(r'(\d{1,2})-$')
+
+
+TAG_RE = re.compile(r'(^.*?)[\|\[$:]', re.MULTILINE)
+ZOOM_RE = re.compile(r'.*?\|z([\d\-]*?)[\[$:]')
+SELECTORS_RE = re.compile(r'(\[.*?\])')
+SUB_RE = re.compile(r'.*:(.*)$')
+ONE_SELECTOR_RE = re.compile(r'\[(.*?)\]')
+
+
+class BlockSplitter:
+    """
+    Should also be initializeable by a preprocessed file
+    """
+    def __init__(self, preprocessed_blocks, write_to_file=False):
+        print("Num of input blocks is: {}".format(len(preprocessed_blocks)))
+        self.blocks = preprocessed_blocks
+        self.split_blocks = {} # selector : list of attributes
+        self.write_to_file = write_to_file
+        self.blocks_by_zoom_level = self.init_blocks_by_zoom_level()
+        print("Will write to file! {}".format(self.write_to_file))
+
+
+    def init_blocks_by_zoom_level(self):
+        ret = {}
+        for i in range(MIN_ZOOM, MAX_ZOOM + 1):
```

Comment:
xrange is better, it does not create sequence.
You can think of it as a lazy sequence.